collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/FormStylingTest.kt

```diff
@@ -28,6 +28,14 @@ class FormStylingTest {
             .assertText("Note text")
     }
 
+    @Test
+    fun requiredQuestionLabelWithHeaderStyleTest() {
+        rule.startAtMainMenu()
+            .copyForm(FORM_FILE_NAME)
+            .startBlankForm(FORM_NAME)
+            .swipeToNextQuestion("Required text question with header style", true)
```

Comment:
We need to assert on the question in the hierarchy as well for this case. Alternatively, we could remove this test and just add a test for `styledQuestionText` which I think might be preferrable.

Reply:
We have a separate test for the hierarchy so I've added it there.

> Alternatively, we could remove this test and just add a test for styledQuestionText which I think might be preferrable.

I was also thinking about it but the fact that method works well doesn't mean we display such styled strings properly. It's tricky and we might for example convert `CharSequence` to `String` after using that method and break it. That's why I preferred to have a UI test.